.github/workflows/python.yml

```diff
@@ -0,0 +1,35 @@
+name: Python Code Quality
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  check-python-code-quality:
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        run: poetry install
+
+      - name: Install ruff
+        run: pip install ruff
```

Comment:
Can this be added to the pyproject.toml so we don't have to have a one off install?